pkgs/unified_analytics/lib/src/initializer.dart

```diff
@@ -9,135 +9,118 @@ import 'package:path/path.dart' as p;
-    required this.homeDirectory,
-    required this.toolsMessageVersion,
-  });
+/// Creates the text file that will contain the client ID
+/// which will be used across all related tools for analytics
+/// reporting in GA.
+void createClientIdFile({required File clientIdFile}) {
+  clientIdFile.createSync(recursive: true);
+  clientIdFile.writeAsStringSync(Uuid().generateV4());
+}
 
-  /// Creates the text file that will contain the client ID
-  /// which will be used across all related tools for analytics
-  /// reporting in GA.
-  static void createClientIdFile({required File clientFile}) {
-    clientFile.createSync(recursive: true);
-    clientFile.writeAsStringSync(Uuid().generateV4());
+/// Creates the configuration file with the default message
+/// in the user's home directory.
+void createConfigFile({
+  required File configFile,
+  required Directory homeDirectory,
+  required FileSystem fs,
+}) {
+  configFile.createSync(recursive: true);
+
+  // If the user was previously opted out, then we will
+  // replace the line that assumes automatic opt in with
+  // an opt out from the start
+  if (legacyOptOut(fs: fs, home: homeDirectory)) {
+    configFile.writeAsStringSync(
+        kConfigString.replaceAll('reporting=1', 'reporting=0'));
+  } else {
+    configFile.writeAsStringSync(kConfigString);
   }
+}
 
-  /// Creates the configuration file with the default message
-  /// in the user's home directory.
-  void createConfigFile({
-    required File configFile,
-    required String dateStamp,
-    required String tool,
-    required int toolsMessageVersion,
-  }) {
-    configFile.createSync(recursive: true);
+/// Creates that file that will persist dismissed survey ids.
+void createDismissedSurveyFile({required File dismissedSurveyFile}) {
+  dismissedSurveyFile.createSync(recursive: true);
+  dismissedSurveyFile.writeAsStringSync('{}');
+}
 
-    // If the user was previously opted out, then we will
-    // replace the line that assumes automatic opt in with
-    // an opt out from the start
-    if (legacyOptOut(fs: fs, home: homeDirectory)) {
-      configFile.writeAsStringSync(
-          kConfigString.replaceAll('reporting=1', 'reporting=0'));
-    } else {
-      configFile.writeAsStringSync(kConfigString);
-    }
-  }
+/// Creates that log file that will store the record formatted
+/// events locally on the user's machine.
+void createLogFile({required File logFile}) {
+  logFile.createSync(recursive: true);
+}
 
-  /// Creates that file that will persist dismissed survey ids.
-  static void createDismissedSurveyFile({required File dismissedSurveyFile}) {
-    dismissedSurveyFile.createSync(recursive: true);
-    dismissedSurveyFile.writeAsStringSync('{}');
-  }
+/// Creates the session file which will contain
+/// the current session id which is the current timestamp.
+///
+/// [sessionIdOverride] can be provided as an override, otherwise it
+/// will use the current timestamp from [Clock.now].
+void createSessionFile({
```

Comment:
What about just returning the `now` value we created, rather than allow a test only override. I prefer as much as possible avoiding having test specific special casing in production code, because it adds complexity, and it also means the what happens in the tests is (at least slightly) different than in prod.

Reply:
The override is not actually for any tests, but for the error handling within the session (now in user_property.dart) functionality.  But either way, I think it makes sense to return `now` instead of passing an override value

Reply:
Ahh yeah, point taken.